simulator/src/bin/generate-example-screenshots.rs

```diff
@@ -0,0 +1,216 @@
+//! Generate example screenshots for the embedded-graphics documentation
+//!
+//! Run from the workspace root with:
+//!
+//! ```bash
+//! cargo run --bin generate-example-screenshots | rustfmt +nightly --config-path rustfmt.nightly.toml
+//! ```
+//!
+//! Screenshots are output to `target/drawing-ops`.
+
+use embedded_graphics::{pixelcolor, prelude::*};
+use embedded_graphics_simulator::{OutputSettingsBuilder, SimulatorDisplay};
+
+const OUTPUT_BASE: &str = "./target/drawing-ops";
+
+macro_rules! op {
+    ($display:ident, $title:expr, $description:expr, $code:block) => {
+        $display.clear(pixelcolor::Rgb888::BLACK).unwrap();
+
+        (|| {
+            $code;
+
+            Ok::<(), core::convert::Infallible>(())
+        })().unwrap();
+
+        let output_settings = OutputSettingsBuilder::new().scale(2).build();
+
+        let path = format!("{}/{}.png", OUTPUT_BASE, $title);
+        $display
+            .to_image_buffer(&output_settings)
+            .save(&path)
+            .unwrap();
+
+        let screenshot = base64::encode(std::fs::read(&path).expect("Couldn't open screenshot"));
+
+        let docs = stringify!($code)
+                .trim_matches(|c| c == '{' || c == '}')
+                .trim();
+
+        // Note: empty lines must remain between HTML elements and inner Markdown for the Markdown
+        // to be parsed correctly.
+        println!(
+            r#"//! ## {}
+            //!
+            //! {}
+            //!
+            //! <div style="display: flex">
+            //! <img style="width: 128px; height: 128px; display: inline-block; margin-right: 8px;" alt="{} example screenshot" src="data:image/png;base64,{}" />
```

Comment:
Why do you use `inline-block` for this? I thought this wouldn't make sense when used with flexbox.